src/main/java/KikuBot.java

```diff
@@ -0,0 +1,199 @@
+
+            if (!parentDir.exists()) {
+                parentDir.mkdirs();
+            }
+
+            FileWriter writer = new FileWriter(file);
+            for (Task task : tasks) {
+                writer.write(task.toFileFormat() + "\n");
+            }
+            writer.close();
+        } catch (IOException e) {
+            System.out.println("An error occurred while saving tasks.");
+            e.printStackTrace();
+        }
+    }
+
+    private static void loadTasks() {
+        File file = new File(FILE_NAME);
+        try {
+            Scanner scanner = new Scanner(file);
+            while (scanner.hasNextLine()) {
+                String line = scanner.nextLine();
+                Task task = parseTask(line);
+                if (task != null) {
+                    tasks.add(task);
+                }
+            }
+            scanner.close();
+        } catch (FileNotFoundException e) {
+            System.out.println("No saved tasks found. Starting with an empty list.");
+            new File(file.getParent()).mkdirs();
+        }
+    }
+
+    private static Task parseTask(String line) {
+        String[] parts = line.split(" \\| ");
+        Task task = null;
+
+        if ("T".equals(parts[0]) && parts.length >= 3) {
+            task = new Todo(parts[2]);
+        } else if ("D".equals(parts[0]) && parts.length >= 4) {
+            task = new Deadline(parts[2], parts[3]);
+        } else if ("E".equals(parts[0]) && parts.length >= 5) {
+            task = new Event(parts[2], parts[3], parts[4]);
+        }
+
+        if (task != null && "1".equals(parts[1])) {
+            task.markAsDone();
+        }
+        return task;
+    }
+
+    private static int markIndex(String userInput, int num) {
+        String markIndexChar = userInput.substring(num, userInput.length());
+        return Integer.parseInt(markIndexChar) - 1;
+    }
+
+    private static Task addTask(String userInput) throws KikuException {
+        int n = userInput.length();
+        if (userInput.startsWith("todo")) {
+            if (userInput.trim().length() == 4) {
+                throw new KikuEmptyTaskException();
+            }
+
+            String description = userInput.substring(5, n);
+            return new Todo(description);
+
+        } else if (userInput.startsWith("deadline")) {
+            if (userInput.trim().length() == 8) {
+                throw new KikuEmptyTaskException();
+            }
+
+            String description = userInput.substring(9, n);
+            if(!description.contains("/by")) {
+                throw new KikuInvalidTaskException("Oh no! Please specify the deadline!");
+            } else if (description.trim().startsWith("/by")) {
+                throw new KikuInvalidTaskException("Oh no! Please specify the description!");
+            }
+
+            String[] deadline = description.split(" /by ");
+            return new Deadline(deadline[0], deadline[1]);
+
```

Comment:
Perhaps you could abstract the cases for this method to shorten the method for improving readability.